src/components/campaignguide/Narrator.tsx

```diff
@@ -0,0 +1,545 @@
+
+    // add all the new tracks after the current track
+    const tracksAfterCurrent = newTracks.slice(currentTrackIndex + 1);
+    await TrackPlayer.add(tracksAfterCurrent);
+  } else {
+    // otherwise reset and add all the new tracks
+    await TrackPlayer.reset();
+    await TrackPlayer.add(newTracks);
+  }
+}
+
+export interface NarrationTrack {
+  id: string;
+  name: string;
+  campaignCode: string;
+  campaignName: string;
+  scenarioName: string;
+}
+
+interface PlayerProps {
+  style?: ViewStyle;
+}
+
+interface PlayerState {
+  track: TrackPlayer.Track | null;
+  state: TrackPlayer.State | null;
+}
+
+class PlayerView extends React.Component<PlayerProps, PlayerState> {
+  onPlaybackTrackChange?: TrackPlayer.EmitterSubscription;
+  onPlaybackState?: TrackPlayer.EmitterSubscription;
+  onPlaybackError?: TrackPlayer.EmitterSubscription;
+
+  constructor(props: PlayerProps) {
+    super(props);
+
+    this.state = {
+      track: null,
+      state: null,
+    };
+  }
+
+  componentDidMount() {
+    this.onPlaybackTrackChange = TrackPlayer.addEventListener(
+      "playback-track-changed",
+      async (data) => {
+        const track = await TrackPlayer.getTrack(data.nextTrack);
+        this.setState({
+          track,
+        });
+      }
+    );
+    this.onPlaybackState = TrackPlayer.addEventListener(
+      "playback-state",
+      async (data) => {
+        this.setState({
+          state: data.state,
+        });
+      }
+    );
+    this.onPlaybackError = TrackPlayer.addEventListener(
+      "playback-error",
+      async (data) => {
+        console.log("playback-error", data);
+      }
+    );
+
+    TrackPlayer.getCurrentTrack()
+      .then((trackId) => TrackPlayer.getTrack(trackId))
+      .then((track) => {
+        this.setState({
+          track,
+        });
+      });
+    TrackPlayer.getState().then((state) => {
+      this.setState({
+        state,
+      });
+    });
+  }
+
+  componentWillUnmount() {
+    this.onPlaybackTrackChange?.remove();
+    this.onPlaybackState?.remove();
+    this.onPlaybackError?.remove();
+  }
+
+  onReplayPress = async () => {
+    try {
+      await TrackPlayer.seekTo((await TrackPlayer.getPosition()) - 30);
+    } catch (_) {}
+  };
+
+  onPreviousPress = async () => {
+    try {
+      await TrackPlayer.skipToPrevious();
+    } catch (_) {}
+  };
+
+  onPlayPress = async () => {
+    const { track, state } = this.state;
+    if (track === null) return;
+
+    if (state === TrackPlayer.STATE_PLAYING) {
+      await TrackPlayer.pause();
+    } else {
+      await TrackPlayer.play();
+    }
+  };
+
+  onClosePress = async () => {
+    await TrackPlayer.stop();
+  };
+
+  onNextPress = async () => {
+    try {
+      await TrackPlayer.skipToNext();
+    } catch (_) {}
+  };
+
+  render() {
+    const { style } = this.props;
+    const { track, state } = this.state;
+
+    return (
+      <TouchableWithoutFeedback style={{ width: "100%" }}>
```

Comment:
Why the TouchableWithoutFeedback here?  Is that to block presses to things undernear the overlay?

Reply:
Ah, that was my experimentation with showing a playlist. I'll remove it